ml/src/main/scala/ai/h2o/sparkling/ml/utils/EstimatorCommonUtils.scala

```diff
@@ -51,4 +52,24 @@ trait EstimatorCommonUtils extends RestCommunication {
     target
   }
 
+  protected def convertModelIdToKey(key: String): String = {
+    if (H2OModel.modelExists(key)) {
+      val replacement = findAlternativeKey(key)
+      logWarning(
```

Comment:
In what scenario does this could happen?

Reply:
Couldn't the suffix be same UUID?

Reply:
I just moved this exact method without any changes from H2OAlgoritm, no changes in the method. 

Not sure if I understand the problem.

Reply:
ok, never mind. I just thought that purpose of this method is to avoid name collisions. I was wondering whether it couldn't be there an alternative approach.

Reply:
Oki, thanks for the explanation. If you see a room for some improvement, please create a JIRA, and let's have a look at it as part of different PR. Thanks!